### PR TITLE
Add GitHub Action workflows

### PR DESCRIPTION
**Pull request by danielpeintner**
Open Questions:
* [ ] failing tests for async-api-converter
* [x] How to deal with eslint and prettier
* [x] in async-api-converter `npm ci` does the build also?
* [x] Do we need the whole test matrix for Windows, MacOS and Ubuntu ?
* [x] How can we test the PR before _merging_?
* [x] Do we want/need Code Coverage as well?

fixes https://github.com/eclipse-thingweb/td-tools/issues/16

### CODE REVIEW

**Comment by egekorkan** [comment]
I will check the tests for async API conversion. For the other questions:

> How to deal with eslint and prettier

We have eslint action at https://github.com/eclipse-thingweb/td-tools/blob/main/.github/eslint.yml but we should create one for prettier.

>  in async-api-converter npm ci does the build also?

The code is javascript so no need for build. We only have a bundling script at https://github.com/eclipse-thingweb/td-tools/blob/main/node/async-api-converter/package.json#L19

> Do we want/need Code Coverage as well?

Why not

> Do we need the whole test matrix for Windows, MacOS and Ubuntu ?

We should discuss this but given that these tools _so far_ are almost pure javascript code (not much to do with Node.js itself), I would say that we need tests only with Ubuntu.

**Comment by danielpeintner** [comment]
@egekorkan and others
I suggest to fix "failing tests for async-api-converter" and look at other aspects maybe also later ...

**Comment by danielpeintner** [comment]
@hasanheroglu improved my draft PR 👍.

I think it is good to be merged. We can still improve in follow-up PRs (code coverage to work might still some *internals* to be fixed)

File: .github/workflows/ci-async-api-converter.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Use Node.js ${{ matrix.os }} ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "npm"
+
+      - name: Install and Build
```
Comment (egekorkan) [inline .github/workflows/ci-async-api-converter.yaml:40]:
```suggestion
      - name: Install Dependencies
```

**Pull request merged**